### PR TITLE
Accounts

### PR DESCRIPTION
**Pull request by jeisses**

### CODE REVIEW

File: contracts/network/network.cpp [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                    {
+                      a.id = id;
+                      a.nonce = 0;
+                      a.address = addr;
+                      a.balance = asset;
+                    });
+  }
+}
+
+void network::transfer_handler(name from, name to, asset quantity, std::string memo) {
```
Comment (laurensV) [inline contracts/network/network.cpp]: Add transfer_handler functionality (deposit into your account when memo is your account id) to README

File: contracts/network/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+```
+open(account_address acc,
+     eosio::extended_symbol symbol,
+     eosio::name payer)
+```
+
+#### transfer
+Transfer tokens between virtual accounts
+```
+transfer(uint64_t from_id,
```
Comment (laurensV) [inline contracts/network/README.md]: function is named `vtransfer`

**Review by laurensV — APPROVED** [review]

**Comment by jdheeter** [comment]
This virtual accounts contract looks really interesting. Is there some more documentation or implementation details? I'd like to learn more about how we could contribute to this and adopt it as well.

**Comment by jeisses** [comment]
> This virtual accounts contract looks really interesting. Is there some more documentation or implementation details? I'd like to learn more about how we could contribute to this and adopt it as well.

Hi John! Thanks for the comment.
We're working on a new documentation site that will explain the inner workings and how to use this system in other apps.

For now the best documentation can be fond in this README and some high level info in my blog post:

- https://github.com/effectai/effect-network/tree/accounts/contracts/vaccount
- https://medium.com/effect-ai/tech-blog-effect-account-system-a3bd8dfe93d4

**Comment by jdheeter** [comment]
Ok great thanks @jeisses 

Do you have an example of the frontend code? To construct/sign the transaction. I'm not sure how the interaction with Metamask works?

**Comment by jeisses** [comment]
We have a new frontend app that works with MetaMask. I'll open source it this week and will link you to some examples then. 

There is a bit of code needed to construct the transactions and to get a correct signature from web3. But not so hard to abstract this away in a few helper functions. 

It would be great if we can get your feedback on this system - I wouldn't mind putting in the effort to make this broadly usable in the EOSIO ecosystem.

**Comment by jdheeter** [comment]
Hey @jeisses would be happy to collaborate with you. Maybe let's setup a call.

**Pull request merged**